spec/integration/api/aspects_controller_spec.rb

```diff
@@ -0,0 +1,302 @@
+      end
+    end
+  end
+
+  describe "#update" do
+    context "with aspect settings" do
+      it "updates full aspect" do
+        new_name = "NewAspectName"
+        new_chat = @aspect2.chat_enabled
+        order = @aspect2.order_id + 1
+        patch(
+          api_v1_aspect_path(@aspect2.id),
+          params: {name: new_name, chat_enabled: new_chat, order: order, access_token: access_token}
+        )
+
+        expect(response.status).to eq(200)
+        aspect = JSON.parse(response.body)
+        expect(aspect["name"]).to eq(new_name)
+        expect(aspect["chat_enabled"]).to eq(new_chat)
+        expect(aspect["order"]).to eq(order)
+        expect(aspect["id"]).to eq(@aspect2.id)
+      end
+
+      it "updates name only aspect" do
+        new_name = "NewAspectName"
+        patch(
+          api_v1_aspect_path(@aspect2.id),
+          params: {name: new_name, access_token: access_token}
+        )
+
+        expect(response.status).to eq(200)
+        aspect = JSON.parse(response.body)
+        expect(aspect["name"]).to eq(new_name)
+        expect(aspect["id"]).to eq(@aspect2.id)
+      end
+
+      it "updates chat only" do
+        new_chat = @aspect2.chat_enabled
+        patch(
+          api_v1_aspect_path(@aspect2.id),
+          params: {chat_enabled: new_chat, access_token: access_token}
+        )
+
+        expect(response.status).to eq(200)
+        aspect = JSON.parse(response.body)
+        expect(aspect["chat_enabled"]).to eq(new_chat)
+        expect(aspect["id"]).to eq(@aspect2.id)
+      end
+
+      it "updates order only" do
+        order = @aspect2.order_id + 1
+        patch(
+          api_v1_aspect_path(@aspect2.id),
+          params: {order: order, access_token: access_token}
+        )
+
+        expect(response.status).to eq(200)
+        aspect = JSON.parse(response.body)
+        expect(aspect["order"]).to eq(order)
+        expect(aspect["id"]).to eq(@aspect2.id)
+      end
+
+      it "succeds with no arguments" do
+        patch(
+          api_v1_aspect_path(@aspect2.id),
+          params: {access_token: access_token}
+        )
+
+        expect(response.status).to eq(200)
+        aspect = JSON.parse(response.body)
+        expect(aspect["name"]).to eq(@aspect2.name)
+        expect(aspect["chat_enabled"]).to eq(@aspect2.chat_enabled)
+        expect(aspect["id"]).to eq(@aspect2.id)
+      end
+    end
+
+    context "with bad parameters" do
+      it "fails with reused name" do
+        patch(
+          api_v1_aspect_path(@aspect2.id),
+          params: {name: @aspect1.name, access_token: access_token}
+        )
+
+        expect(response.status).to eq(422)
+        expect(response.body).to eq(I18n.t("api.endpoint_errors.aspects.cant_update"))
+      end
+
+      it "fails with bad id" do
+        patch(
+          api_v1_aspect_path("-1"),
+          params: {name: "NewAspectName", access_token: access_token}
+        )
+
+        expect(response.status).to eq(422)
```

Comment:
Maybe that 404 would be more accurate.

Reply:
I believe you are correct.  Let me look into that.